### ISSUE TITLE
--include option not working 

### ISSUE THREAD

**Issue by SrikanthShyamanur** [nexB/scancode-toolkit#2059]
### Description

**_--include_** option is not working.

### How To Reproduce

Run scancode on samples folder with **_--include_** option

### Output :
`Setup plugins...`
`Collect file inventory...`
`Scanning done.`
`Summary:         with 1 process(es)`
`Errors count:   0`
`Scan Speed:     0.00 files/sec.`
`Initial counts: 44 resource(s): 33 file(s) and 11 directorie(s)`
`Final counts:   1 resource(s): 0 file(s) and 1 directorie(s)`
`Timings:`
`  scan_start: 2020-06-03T123841.356864`
`  scan_end:   2020-06-03T123841.401388`

> For bug reports, it really helps us to know:

* What OS are you running on? Linux
* What version of scancode-toolkit was used to generate the scan file?  v3.1.1
* What installation method was used to install/run scancode? source download

**Comment by pombredanne** [comment]
Good catch! that's a bug alright. Thank you

**Comment by SrikanthShyamanur** [comment]
Seams same issue for "--ignore", please check

**Comment by SrikanthShyamanur** [comment]
Is there a version in which this feature is working ?

**Comment by pombredanne** [comment]
`--ignore` seem to work fine in develop:
```
$ ./scancode samples/JGroups --json-pp - --ignore "*.java"
Setup plugins...
Collect file inventory...
{
  "headers": [
    {
      "tool_name": "scancode-toolkit",
      "tool_version": "3.1.2",
      "options": {
        "input": [
          "samples/JGroups"
        ],
        "--ignore": [
          "*.java"
        ],
        "--json-pp": "-"
      },
      "notice": "Generated with ScanCode and provided on an \"AS IS\" BASIS, WITHOUT WARRANTIES\nOR CONDITIONS OF ANY KIND, either express or implied. No content created from\nScanCode should be considered or used as legal advice. Consult an Attorney\nfor any legal advice.\nScanCode is a free software code scanning tool from nexB Inc. and others.\nVisit https://github.com/nexB/scancode-toolkit/ for support and download.",
      "start_timestamp": "2020-06-03T204621.358905",
      "end_timestamp": "2020-06-03T204621.366012",
      "duration": 0.007116556167602539,
      "message": null,
      "errors": [],
      "extra_data": {
        "files_count": 7
      }
    }
  ],
  "files": [
    {
      "path": "JGroups",
      "type": "directory",
      "scan_errors": []
    },
    {
      "path": "JGroups/EULA",
      "type": "file",
      "scan_errors": []
    },
    {
      "path": "JGroups/LICENSE",
      "type": "file",
      "scan_errors": []
    },
    {
      "path": "JGroups/src",
      "type": "directory",
      "scan_errors": []
    },
    {
      "path": "JGroups/licenses",
      "type": "directory",
      "scan_errors": []
    },
    {
      "path": "JGroups/licenses/apache-1.1.txt",
      "type": "file",
      "scan_errors": []
    },
    {
      "path": "JGroups/licenses/apache-2.0.txt",
      "type": "file",
      "scan_errors": []
    },
    {
      "path": "JGroups/licenses/bouncycastle.txt",
      "type": "file",
      "scan_errors": []
    },
    {
      "path": "JGroups/licenses/cpl-1.0.txt",
      "type": "file",
      "scan_errors": []
    },
    {
      "path": "JGroups/licenses/lgpl.txt",
      "type": "file",
      "scan_errors": []
    }
  ]
}
Scanning done.
Summary:         with 1 process(es)
Errors count:   0
Scan Speed:     0.00 files/sec. 
Initial counts: 17 resource(s): 14 file(s) and 3 directorie(s) 
Final counts:   10 resource(s): 7 file(s) and 3 directorie(s) 
Timings:
  scan_start: 2020-06-03T204621.358905
  scan_end:   2020-06-03T204621.366012
Removing temporary files...done.
```

**Comment by pombredanne** [comment]
FWIW, here are the tests on ignore 
https://github.com/nexB/scancode-toolkit/blob/develop/tests/scancode/test_plugin_ignore.py

**Comment by pombredanne** [comment]
And there are no tests for the --include :|

**Comment by pombredanne** [comment]
And it seems like the underlying function has no tests for includes either: https://github.com/nexB/scancode-toolkit/blob/develop/tests/commoncode/test_ignore.py

**Comment by SrikanthShyamanur** [comment]
Is this issue being considered for next release ?

**Comment by pombredanne** [comment]
It sure could be, but some help would be mucho welcomed there.
My feeling is that the feature needs to be reset by using something that strictly matches the .gitignore ways, either by testing and fixing it here or by adopting a new library for that.
Some possible inputs:
- @mherrmann https://github.com/mherrmann/gitignore_parser is maintained, tested and has clean code.  
- several other such as https://github.com/NiklasRosenstein/python-nr/tree/master/nr.gitignore or  https://github.com/NiklasRosenstein-Python/nr-deprecated/blob/master/nr/gitignore.py abandoned

**Comment by xiao623** [comment]
I've got this problem, too. 
Is there any new progress?

**Comment by pombredanne** [comment]
@tardyp  kindly fixed this in #2395  :bow:
